src/ziggurat/streams.clj

```diff
@@ -156,11 +156,29 @@
   (.transformValues stream-builder (header-transformer-supplier) (into-array [(.name (store-supplier-builder))])))
 
 (declare stream)
+(def restarted-streams (atom {}))
```

Comment:
Can we not use it please!

Reply:
Mount might not be flexible enough for this use case.

Reply:
@mjayprateek please add what all tests should pass for this to get merged?

Reply:
**Test Cases**

Test Case 1
- start all the streams using `mount/start`. Verify that streams for all the configs were started
- stop the streams using `mount/stop`. Verify that streams for all the configs were stopped

Test Case 2
- Start all the streams using mount/start
- Stop one of the streams using the new method. Verify that the stream has stopped
- Restart this stream with the new method. Verify that the atom contains this new stream object
- Stop all the streams using mount/stop. Verify that all the streams including the new one were stopped

Test Case 3
- Start all the streams using mount/start
- Use `start-stream` (the new method) to start a duplicate stream for a given topic-entity
- Verify that no new stream object is constructed or no new stream threads are started.
- mount/stop should close all the stream objects

Test Case 4
- Start all the streams using mount/start
- stop a stream using the new method
- start a stream using the new method
- stop all the streams using mount/stop
- [IMP] Verify that the atom `restarted-streams` is empty
- Additionally, try verifying that once mount/stop is called,  neither of the new methods (`stop-stream` and `restart-stream`) do anything.

Reply:
added